### ISSUE TITLE
Java exception in version 2.3.0 with Pycharm 2016.1.5 

### ISSUE THREAD

**Issue by jalopez** [vsch/idea-multimarkdown#362]
After installing plugin version 2.3.0 in Pycharm 2016.1.5, the following exception is raised continuously

```
java.lang.NoClassDefFoundError: com/intellij/openapi/util/Setter
java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: com/intellij/openapi/util/Setter
	at com.vladsch.idea.multimarkdown.language.folding.MarkdownCodeFoldingOptionsProvider.<init>(MarkdownCodeFoldingOptionsProvider.java:27)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.picocontainer.defaults.InstantiatingComponentAdapter.newInstance(InstantiatingComponentAdapter.java:193)
	at org.picocontainer.defaults.ConstructorInjectionComponentAdapter$1.run(ConstructorInjectionComponentAdapter.java:220)
	at org.picocontainer.defaults.ThreadLocalCyclicDependencyGuard.observe(ThreadLocalCyclicDependencyGuard.java:53)
	at org.picocontainer.defaults.ConstructorInjectionComponentAdapter.getComponentInstance(ConstructorInjectionComponentAdapter.java:248)
	at com.intellij.util.pico.ConstructorInjectionComponentAdapter.getComponentInstance(ConstructorInjectionComponentAdapter.java:59)
	at com.intellij.openapi.extensions.AbstractExtensionPointBean.instantiate(AbstractExtensionPointBean.java:75)
	at com.intellij.openapi.options.ConfigurableEP$ClassProducer.createElement(ConfigurableEP.java:347)
	at com.intellij.openapi.options.ConfigurableEP.createConfigurable(ConfigurableEP.java:273)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.a(ConfigurableWrapper.java:54)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.wrapConfigurable(ConfigurableWrapper.java:49)
	at com.intellij.openapi.options.ex.ConfigurableWrapper$1.fun(ConfigurableWrapper.java:68)
	at com.intellij.openapi.options.ex.ConfigurableWrapper$1.fun(ConfigurableWrapper.java:65)
	at com.intellij.util.containers.ContainerUtil.mapNotNull(ContainerUtil.java:1889)
	at com.intellij.util.containers.ContainerUtil.mapNotNull(ContainerUtil.java:1838)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.createConfigurables(ConfigurableWrapper.java:65)
	at com.intellij.application.options.editor.CodeFoldingConfigurable.createConfigurables(CodeFoldingConfigurable.java:113)
	at com.intellij.openapi.options.CompositeConfigurable.getConfigurables(CompositeConfigurable.java:61)
	at com.intellij.application.options.editor.CodeFoldingConfigurable.createComponent(CodeFoldingConfigurable.java:61)
	at com.intellij.ide.ui.ConfigurableOptionsTopHitProvider.getOptions(ConfigurableOptionsTopHitProvider.java:126)
	at com.intellij.ide.ui.OptionsTopHitProvider.consumeTopHits(OptionsTopHitProvider.java:60)
	at com.intellij.ide.util.gotoByName.GotoActionItemProvider.c(GotoActionItemProvider.java:118)
	at com.intellij.ide.util.gotoByName.GotoActionItemProvider.filterElements(GotoActionItemProvider.java:82)
	at com.intellij.ide.actions.SearchEverywhereAction$CalcThread.a(SearchEverywhereAction.java:1539)
	at com.intellij.ide.actions.SearchEverywhereAction$CalcThread.i(SearchEverywhereAction.java:1570)
	at com.intellij.ide.actions.SearchEverywhereAction$CalcThread.run(SearchEverywhereAction.java:1452)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:369)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by vsch** [comment]
@jalopez, thank you for reporting this. I updated the implementation to the new API without thinking. This API is not available in earlier releases. I rolled back the change.

EAP update with a fix for the issue is available. To enable EAP update channel:

![Screen Shot Settings Update Channel](https://github.com/vsch/idea-multimarkdown/raw/master/assets/images/faq/ScreenShot_Settings_UpdateChannel.png)

Then make the IDE check for updates.
